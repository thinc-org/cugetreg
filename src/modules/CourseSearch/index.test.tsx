import { shallow } from 'enzyme'

import React from 'react'

describe('CourseSearchPage', () => {
  const mockOpenFilterBar = false
  const mockAcademicYear = '2564'
  const mockSemester = '1'

  const setTermSpy = jest.fn()
  const setOpenFilterBarSpy = jest.fn()
  const useCourseSearchPageSpy = jest.fn(() => ({
    openFilterBar: mockOpenFilterBar,
    setOpenFilterBar: setOpenFilterBarSpy,
    onOpen: jest.fn(),
  }))

  jest.doMock('./hooks/useCourseSearchPage', () => ({
    useCourseSearchPage: useCourseSearchPageSpy,
  }))
  jest.doMock('@/services/apollo', () => ({
    initializeApollo: jest.fn(),
  }))
  jest.doMock('@/common/hooks/useCourseGroup', () => ({
    useCourseGroup: () => ({
      academicYear: mockAcademicYear,
      semester: mockSemester,
      setTerm: setTermSpy,
    }),
  }))

  it('Should match snapshot correctly', async () => {
    const { CourseSearchPage } = await import('.')

    const wrapper = shallow(<CourseSearchPage />)
    expect(wrapper).toMatchSnapshot()
  })
})
