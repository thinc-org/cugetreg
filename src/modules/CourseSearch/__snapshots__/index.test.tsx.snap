// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseSearchPage Should match snapshot correctly 1`] = `
<Styled(div)>
  <PageMeta
    title="ค้นหาวิชาเรียน"
  />
  <Styled(Component)
    alignItems="center"
    direction="row"
    justifyContent="space-between"
    spacing={2}
  >
    <ForwardRef(Typography)
      variant="h2"
    >
      ค้นหาวิชาเรียน
    </ForwardRef(Typography)>
    <Hidden
      mdUp={true}
    >
      <Analytics
        elementName="open_shopping_cart_button"
      >
        <Memo(wrappedComponent)
          onClick={[MockFunction]}
        />
      </Analytics>
    </Hidden>
  </Styled(Component)>
  <Styled(Component)
    alignItems="flex-start"
  >
    <Styled(Component)
      direction="row"
      spacing={2}
      width="100%"
    >
      <SearchField />
      <Analytics
        elementName="filter_button"
      >
        <FilterIconButton />
      </Analytics>
      <Hidden
        mdDown={true}
      >
        <Analytics
          elementName="selected_courses_button"
        >
          <Memo(wrappedComponent)
            onClick={[MockFunction]}
          />
        </Analytics>
      </Hidden>
    </Styled(Component)>
    <TagList />
  </Styled(Component)>
  <NoTagListLayout />
  <Styled(Component)
    direction="row"
    spacing={3}
  >
    <CourseList />
    <FilterSection
      open={false}
    />
  </Styled(Component)>
</Styled(div)>
`;
